main.js

```diff
@@ -0,0 +1,256 @@
+"use strict";
+
+const {context} = require("@arangodb/locals");
+const createRouter = require("@arangodb/foxx/router");
+const {query, db} = require("@arangodb");
+const graph_module = require("@arangodb/general-graph");
+const queues = require("@arangodb/foxx/queues");
+const joi = require("joi");
+const {modelTypes, metadataCollectionName} = require("./scripts/setup");
+
+const router = createRouter();
+context.use(router);
```

Comment:
I'd move the entire router into a separate file (like `api/model.js` or something) and then import it from there to mount it here but that's a matter of taste. It makes it easier to split up code later though, especially if you have routes that are mostly unrelated.

Reply:
I've refactored it to split out the router and the rest of the logic that was in `main.js` into a couple of separate chunks under `api/`